packages/protocol/test-sol/unit/common/Registry.t.sol

```diff
@@ -14,89 +14,111 @@ contract RegistryTest is Test {
   // hash is harcoded to avoid test and implementation changing at the same time
   bytes32 constant ID_HASH = 0x05445421d7b4d4c2e571c5a4ccf9317ec68601449f752c75ddbcc61a16061004;
 
-  Registry registry;
+  Registry _registry;
   address owner;
 
   function setUp() public {
     owner = address(this);
     vm.prank(owner);
-    registry = new Registry(true);
-    registry.initialize();
+    _registry = new Registry(true);
+    _registry.initialize();
   }
+}
+
+contract RegistryTest_L2 is WhenL2, RegistryTest {
+  function setUp() public {
+    super.setUp();
+    registry = IRegistry(address(_registry));
```

Comment:
would move this to the parent and get rid of the super

Reply:
actually, why do we need to instances Registry (_registry and registry). I know one is the contract and another an interface, but why we need both?

Reply:
This is only done here because I need to set the registry address for the instance initialized in the Utils contract. This is only the case for L2